File: xijinhui_window/XiJinHuiWindow/Models/DiningTable.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Text.Json.Serialization;
using Telerik.WinControls.UI;
using static System.Windows.Forms.VisualStyles.VisualStyleElement;
using XiJinHuiWindow.Utils;

namespace XiJinHuiWindow.Models
{
    public class DiningTable
    {
        public static Dictionary<int, string> FlagStatuses = new Dictionary<int, string>
        {
            //状态 1:空闲 2:使用中 3:预定 4:清洁 5:维修
            //1: Idle 2: In use 3: Reserved 4: Cleaning 5: Repair
             { 1, "Idle"},
             { 2, "InUse"},
             { 3, "Reserved"},
             { 4, "Cleaning"},
             { 5, "Repair"},
        };
            // Root myDeserializedClass = JsonConvert.DeserializeObject<Root>(myJsonResponse);
            [JsonPropertyName("id")]
            public int? Id { get; set; }

            [JsonPropertyName("dining_table_code")]
            public string DiningTableCode { get; set; }
        
            public string DiningFloorName { get; set; }

            [JsonPropertyName("english_name")]
            public string EnglishName { get; set; }

            [JsonPropertyName("chinese_name")]
            public string ChineseName { get; set; }

            [JsonPropertyName("khmer_name")]
            public string KhmerName { get; set; }

            [JsonPropertyName("remark")]
            public string Remark { get; set; }

            [JsonPropertyName("person_count")]
            public int? PersonCount { get; set; }

            [JsonPropertyName("tax_percentage"), JsonConverter(typeof(StringToDoubleConverter))]
            public double? TaxPercentage { get; set; }

            [JsonPropertyName("tax_fixPrice"), JsonConverter(typeof(StringToDoubleConverter))]
            public double? TaxFixPrice { get; set; }

            [JsonPropertyName("room_service_percentage"), JsonConverter(typeof(StringToDoubleConverter))]
            public double? RoomServicePercentage { get; set; }

            [JsonPropertyName("room_service_fixPrice"), JsonConverter(typeof(StringToDoubleConverter))]
            public double? RoomServiceFixPrice { get; set; }

            [JsonPropertyName("is_broken")]
            public bool? IsBroken { get; set; }

            [JsonPropertyName("is_enable")]
            public bool? IsEnable { get; set; }

            [JsonPropertyName("flag_status")]
            public int? FlagStatus { get; set; }

            [JsonPropertyName("created_at")]
            public string CreatedAt { get; set; }

            [JsonPropertyName("updated_at")]
            public string UpdatedAt { get; set; }

            [JsonPropertyName("dining_floor_id")]
            public int? DiningFloorId { get; set; }

            [JsonPropertyName("dining_floor")]
            public DiningFloor DiningFloor { get; set; }

    }
}

File: xijinhui_flutter/lib/app/data/models/dining_table.dart
class DiningTable {
  int? id;
  String? diningTableCode;
  String? englishName;
  String? chineseName;
  String? khmerName;
  String? remark;
  int? personCount;
  double? taxPercentage;
  double? taxFixPrice;
  double? roomServicePercentage;
  double? roomServiceFixPrice;
  bool? isBroken;
  bool? isEnable;
  int? calculationType;
  int? flagStatus;
  int? diningFloorId;

  DiningTable(
      {this.id,
      this.diningTableCode,
      this.englishName,
      this.chineseName,
      this.khmerName,
      this.remark,
      this.personCount,
      this.taxPercentage,
      this.taxFixPrice,
      this.roomServicePercentage,
      this.roomServiceFixPrice,
      this.isBroken,
      this.isEnable,
      this.calculationType,
      this.flagStatus,
      this.diningFloorId});

  DiningTable.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    diningTableCode = json['dining_table_code'];
    englishName = json['english_name'];
    chineseName = json['chinese_name'];
    khmerName = json['khmer_name'];
    remark = json['remark'];
    personCount = json['person_count'];
    taxPercentage = double.tryParse("${json['tax_percentage'] ?? 0}");
    taxFixPrice = double.tryParse("${json['tax_fixPrice'] ?? 0}");
    roomServicePercentage =
        double.tryParse("${json['room_service_percentage'] ?? 0}");
    roomServiceFixPrice =
        double.tryParse("${json['room_service_fixPrice'] ?? 0}");
    isBroken = bool.tryParse("${json['is_broken'] ?? ''}");
    isEnable = bool.tryParse("${json['is_enable'] ?? ''}");
    calculationType = json['calculation_type'];
    flagStatus = json['flag_status'];
    diningFloorId = json['dining_floor_id'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['dining_table_code'] = diningTableCode;
    data['english_name'] = englishName;
    data['chinese_name'] = chineseName;
    data['khmer_name'] = khmerName;
    data['remark'] = remark;
    data['person_count'] = personCount;
    data['tax_percentage'] = taxPercentage;
    data['tax_fixPrice'] = taxFixPrice;
    data['room_service_percentage'] = roomServicePercentage;
    data['room_service_fixPrice'] = roomServiceFixPrice;
    data['is_broken'] = isBroken;
    data['is_enable'] = isEnable;
    data['calculation_type'] = calculationType;
    data['flag_status'] = flagStatus;
    data['dining_floor_id'] = diningFloorId;
    return data;
  }
}
